lang/src/main/kotlin/org/partiql/lang/eval/io/DelimitedValues.kt

```diff
@@ -50,24 +49,24 @@ object DelimitedValues {
     enum class ConversionMode {
         /** Attempt to parse each value as a scalar, and fall back to string. */
         AUTO {
-            override fun convert(valueFactory: ExprValueFactory, raw: String): ExprValue = try {
-                val converted = valueFactory.ion.singleValue(raw)
+            override fun convert(ion: IonSystem, raw: String): ExprValue = try {
+                val converted = ion.singleValue(raw)
                 when (converted) {
                     is IonInt, is IonFloat, is IonDecimal, is IonTimestamp ->
-                        valueFactory.newFromIonValue(converted)
+                        ExprValue.of(converted)
                     // if we can't convert the above, we just use the input string as-is
-                    else -> valueFactory.newString(raw)
+                    else -> ExprValue.newString(raw)
```

Comment:
I think generally we should migrate from `IonValue` to `IonElement`, and once we are doing this, `IonSystem` will disappear anyway.

Reply:
And for the `DelimitedValues`, it is only used in cli/ folder now. I think we should migrate it to cll/ folder, but since it is public right now,  it might take some effort to deprecate it, considering backward compatibility.

Reply:
So my point of view is that let's just keep the `IonSystem` there. We can consider this again when we are doing the work migrating from `IonValue` to `IonElement`.

Reply:
There is value in simplifying the signature of `convert()` right now, before the eventual transition to `IonElement`. 

Also, see my another comment about factoring out the places where IonSystem (and then IonElement) are only needed for "parsing" into a native JVM type.  I hope neither IonSystem nor IonElement are needed for this purpose.

Reply:
Whether `DelimitedValues` is migrated to cli or not does not affect the matter at hand, does it?

Reply:
For backward compatibility concern, we can write a new internal `DelimitedValue` in cli/ folder and mark this one as deprecated.

Reply:
Please do not relocate `DelimitedValues` in this PR.  There can be an argument where is a better home for it, but it is not in one specific tool (cli), since it is a fairly generic i/o functionality.  It is already in a clearly marked `io` package, so this is not too bad. 

`convert()` signature is already being changed, so I merely suggest to make it simpler.  Whether to worry about backward compatibility or not is an orthogonal matter. I'd dare say "no", but if "yes" -- by the same means of introducing a 2nd method.

Reply:
Ok, I will deprecate the original method and add a new one without `ExprValueFactory`.